Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py

```diff
@@ -561,12 +568,19 @@ def PyExec(self):
                         self._apply_transmission(ws, transmission_ws)
                     solid_angle = self._make_solid_angle_name(ws)
                     cache = self.getProperty('CacheSolidAngle').value
-                    if cache:
-                        if not mtd.doesExist(solid_angle):
-                            SolidAngle(InputWorkspace=ws, OutputWorkspace=solid_angle,
-                                       Method=self._get_solid_angle_method(self._instrument))
-                    else:
-                        SolidAngle(InputWorkspace=ws, OutputWorkspace=solid_angle,
+                    if (cache and not mtd.doesExist(solid_angle)) or not cache:
+                        if self._instrument == "D16":
+                            run = mtd[ws].getRun()
+                            distance = run.getLogData("Det.value").value/1000
+                            CloneWorkspace(InputWorkspace=ws, OutputWorkspace=solid_angle)
+                            MoveInstrumentComponent(Workspace=solid_angle, X=0, Y=0, Z=distance,
```

Comment:
Why do you need to move? The ws is already at the right distance, I think you just need to rotate, no?

Reply:
Since SolidAngle uses the Rectangle method, it is theta dependent, theta being calculated with regard to the beam. If it is not moved, I don't think the calculated solid angle will be correct.